### PR TITLE
Enable python3.12 in GHA

### PR DESCRIPTION
**Pull request by akuporos**
### Details:
 - *item1*
 - *...*

### Tickets:
 - CVS-149210

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
We typically use native python version for Linux OSes.
I suppose other python versions like 3.12 will be added separately within activity of aligning GHA / Jenkins workflows.

File: .github/workflows/send_workflows_to_opentelemetry.yml [hunk 1]
```diff
@@ -9,21 +9,21 @@ on:
       - Code snippets
       - Code Style
       - Code coverage
-      - Coverity (Ubuntu 20.04, Python 3.11)
+      - Coverity (Ubuntu 20.04, Python 3.12)
       - Fedora (RHEL), Python 3.9
       - Linux (Ubuntu 20.04, Python 3.8)
```
Comment (rkazants) [inline .github/workflows/send_workflows_to_opentelemetry.yml:14]:
```suggestion
      - Linux (Ubuntu 20.04, Python 3.12)
```

**Comment by rkazants** [comment]
> We typically use native python version for Linux OSes. I suppose other python versions like 3.12 will be added separately within activity of aligning GHA / Jenkins workflows.

Agree, the default version of Python must be validated as well.

**Comment by p-wysocki** [comment]
In that case how should we proceed with testing Python 3.12 in GHA? Back when we had Azure we bumped Python version in all jobs, in addition to usual extension of Jenkins pipelines.

 - https://github.com/openvinotoolkit/openvino/pull/15144
 - https://github.com/openvinotoolkit/openvino/pull/12578

Now the situation has changed, and also I suppose GHA cover more code than Azure did. In that case should we:

 - Bump only some GHA jobs to Python 3.12?
 - Introduce new jobs or somehow parametrize existing ones to run Python version checks sequentially like Jenkins does?
 - Keep Python 3.12 testing in Jenkins, as GHA are already testing a single Python version?

cc @akuporos @almilosz @ilya-lavrenov @rkazants

**Comment by ilya-lavrenov** [comment]
I suggest waiting for "official" GHA work which will align Jenkins and GHA pipelines in terms of Python versions coverage as well.

CC @dorloff

**Comment by akuporos** [comment]
Close it in favor https://github.com/openvinotoolkit/openvino/pull/26466

**Pull request closed without merging**